drivers/platform/mbed/mbed_spi.cpp

```diff
@@ -0,0 +1,280 @@
+#ifdef __cplusplus
+extern "C"
+{
+#endif //  _cplusplus
+
+#include "no-os/error.h"
+#include "no-os/spi.h"
+#include "no-os/gpio.h"
+#include "mbed_spi.h"
+
+#define		SPI_8_BIT_FRAME			8		// SPI 8-bit frame size
+
+/******************************************************************************/
+/************************ Functions Definitions *******************************/
+/******************************************************************************/
+
+/**
+ * @brief Initialize the Mbed SPI communication peripheral.
+ * @param desc[in,out] - The SPI descriptor.
+ * @param param[in] - The structure that contains the SPI parameters.
+ * @return SUCCESS in case of success, negative error code otherwise.
+ */
+int32_t mbed_spi_init(struct spi_desc **desc,
+		      const struct spi_init_param *param)
+{
+	mbed::SPI *spi;					// Pointer to new spi instance
+	DigitalOut *csb;  				// pointer to new CSB gpio instance
+	struct mbed_spi_desc *mbed_spi_desc;	// Pointer to mbed spi descriptor
+	use_gpio_ssel_t use_gpio_ssel;	// For CSB hardware (implicit) control
+	struct spi_desc *spi_desc;
+
+	if (!desc || !param)
+		return -EINVAL;
+
+	/* Allocate memory for spi_desc structure */
+	spi_desc = (struct spi_desc *)calloc(1, sizeof(*spi_desc));
+	if (!spi_desc)
+		return -ENOMEM;
+
+	spi_desc->chip_select = param->chip_select;
+	spi_desc->mode = param->mode;
+	spi_desc->max_speed_hz = param->max_speed_hz;
+
+	/* Allocate memory for mbed_spi_desc structure */
+	mbed_spi_desc = (struct mbed_spi_desc *)calloc(1, sizeof(*mbed_spi_desc));
+	if (!mbed_spi_desc)
+		goto err_mbed_spi_desc;
+
+	/* Configure and instantiate SPI peripheral */
+	if (((struct mbed_spi_init_param *)param->extra)->use_sw_csb) {
+		/* CSB pin toggled explicitly (s/w controlled) during SPI transaction */
+		spi = new SPI(
+			(PinName)(((struct mbed_spi_init_param *)param->extra)->spi_mosi_pin),
+			(PinName)(((struct mbed_spi_init_param *)param->extra)->spi_miso_pin),
+			(PinName)(((struct mbed_spi_init_param *)param->extra)->spi_clk_pin));
+
+		if (spi) {
+			/* Configure and instantiate chip select pin */
+			csb = new DigitalOut((PinName)(spi_desc->chip_select));
+			if (csb) {
+				mbed_spi_desc->use_sw_csb = true;
+				mbed_spi_desc->csb_gpio = (DigitalOut *)csb;
+				csb->write(GPIO_HIGH);
+			} else {
+				goto err_csb;
+			}
+		}
+	} else {
+		/* CSB pin toggled implicitly (through HAL layer) during SPI transaction */
+		spi = new SPI(
+			(PinName)(((struct mbed_spi_init_param *)param->extra)->spi_mosi_pin),
+			(PinName)(((struct mbed_spi_init_param *)param->extra)->spi_miso_pin),
+			(PinName)(((struct mbed_spi_init_param *)param->extra)->spi_clk_pin),
+			(PinName)(param->chip_select),
+			use_gpio_ssel);
+
+		mbed_spi_desc->use_sw_csb = false;
+		mbed_spi_desc->csb_gpio = NULL;
+	}
+
+	if (!spi) {
+		goto err_spi;
+	}
+
+	mbed_spi_desc->spi_port = (SPI *)spi;
+
+	spi_desc->extra = (struct mbed_spi_desc *)mbed_spi_desc;
```

Comment:
These pointers should be cast to `void *` to be allocated to a void pointer.But I think we can skip casting here.

Reply:
Make sense. Removed casting here.